dataprep-pipelines/tests/integration.rs

```diff
@@ -1,44 +1,80 @@
+#[macro_use]
+extern crate lazy_static;
+extern crate rand;
+
 use dir_assert::assert_paths;
+use rand::Rng;
 use std::path::PathBuf;
 use tokio::fs;
+use tokio::io::AsyncWriteExt;
 
-#[tokio::test]
-async fn it_works_for_one_file() {
+lazy_static! {
+    static ref MANIFEST_FILE: PathBuf = PathBuf::from("test/manifest.json");
+    static ref TEST_DIR: PathBuf = PathBuf::from("test");
+    static ref INPUT_DIR: PathBuf = PathBuf::from("test/input");
+    static ref OUTPUT_DIR: PathBuf = PathBuf::from("test/output");
+    static ref UNPACKED_DIR: PathBuf = PathBuf::from("test/unpacked");
+}
+
+async fn setup_structure() {
     // remove any old test crud
-    fs::remove_dir_all("test").await.unwrap();
-    fs::create_dir("test").await.unwrap();
+    fs::remove_dir_all(&*TEST_DIR).await.unwrap();
+    fs::create_dir(&*TEST_DIR).await.unwrap();
     // create input directory
-    fs::create_dir("test/input").await.unwrap();
-    let input_dir = PathBuf::from("test/input");
+    fs::create_dir(&*INPUT_DIR).await.unwrap();
     // create output directory
-    fs::create_dir("test/output").await.unwrap();
-    let output_dir = PathBuf::from("test/output");
+    fs::create_dir(&*OUTPUT_DIR).await.unwrap();
     // create final output directory for unpacked
-    fs::create_dir("test/unpacked").await.unwrap();
-    let unpacked_dir = PathBuf::from("test/unpacked");
-    let manifest_file = PathBuf::from("test/manifest.json");
-    // create a file in the input directory
-    fs::write("test/input/test.txt", b"test").await.unwrap();
+    fs::create_dir(&*UNPACKED_DIR).await.unwrap();
+}
 
+async fn transform_and_check() {
     // run the function
     println!("doing pack pipeline!");
     dataprep_pipelines::pipeline::pack_pipeline::pack_pipeline(
-        input_dir.clone(),
-        output_dir.clone(),
-        manifest_file.clone(),
+        INPUT_DIR.clone(),
```

Comment:
soon i'm going to ask you to remove like 90% of these pathbuf "clones" and make sure we're using all pointers to the same underlying data as much as possible

Reply:
but not yet- can you make sure there's an issue for that though before we merge this?